lib/Core/Utils.cpp

```diff
@@ -36,5 +39,40 @@ namespace CIL {
                     matrix[i][j] *= constant;
         }
 
+        static double computeOverlappingArea(Pixel px, Coordinate S)
+        {
+            double left = 0, right = 0, up = 0, down = 0;
+            left = std::max(static_cast<double>(px.col()), S.x - 0.5);
+            right = std::min(static_cast<double>(px.col() + 1), S.x + 0.5);
+            up = std::max(static_cast<double>(px.row()), S.y - 0.5);
+            down = std::min(static_cast<double>(px.row() + 1), S.y + 0.5);
+            if (left > right || up > down)
+                return 0;
+            double res = (right - left) * (down - up);
+            return res;
+        }
+
+        DetachedFPPixel bilinearInterpolation(ImageInfo& img_info,
+                                              Coordinate source)
+        {
+            std::pair<int, int> x_range;
+            std::pair<int, int> y_range;
+            x_range = {source.x - 0.5, source.x + 0.5};
+            y_range = {source.y - 0.5, source.y + 0.5};
+            DetachedFPPixel dpx(img_info.numComponents());
+
+            for (auto i = y_range.first; i <= y_range.second; ++i)
+            {
+                for (auto j = x_range.first; j <= x_range.second; ++j)
```

Comment:
I think this could be done using bounded pixel (`px.setBounds`), would be interested in making that change?

Reply:
Yes, I would be interested in making this change. But here, values of x ranges and y ranges can also be negative (basically invalid range), should `setBounds` handle this as well?

Reply:
I think the constructor for `Dimensions(origin, length, breadth)` ( which will be passed to pixel ) truncates the dimensions so that it remains inside the image. It is untested but can be patched if it doesn't work as intended. You can try using that

Reply:
I will try using `setBounds`:). 
Just for further clarifications, in this particular code snippet, the maximum difference in the ranges will be equal to 1.